theano/tensor/elemwise.py

```diff
@@ -2041,6 +2061,7 @@ class Prod(CAReduceDtype):
     input.
 
     """
+    __props__ = ("axis", "dtype", "acc_dtype")
```

Comment:
I won't repeat it. In this file, check if you need to remove \__eq__ and \__hash__ for each class you added \__props__. Take the time to understand if you need to remove them or not and only do it if it is needed.